source/core/Variants.cpp

```diff
@@ -26,53 +26,76 @@ SDG
 namespace Vireo
 {
 
-bool VariantType::IsNullVariant(VariantTypeRef variant)
+//------------------------------------------------------------
+VariantType::VariantType(TypeManagerRef typeManager)
+    : TypeCommon(typeManager)
 {
-    return variant == nullptr;
+    _isFlat = false;
+    _ownsDefDefData = false;
+    _isMutableValue = true;
+    _hasCustomDefault = false;
+    _topAQSize = TypeManager::HostPointerToAQSize();
+    _aqAlignment = TypeManager::AQAlignment(sizeof(void*));
+    _encoding = kEncoding_Variant;
 }
 
-bool VariantType::IsNullOrEmptyVariant(VariantTypeRef variant)
+NIError VariantType::InitData(void* pData, TypeRef pattern)
 {
-    return variant == nullptr
-        || (variant->_underlyingTypeRef->BitEncoding() == kEncoding_None  // variant's data is empty
-            && IsAttributeMapNullOrEmpty(variant));  // variant's attributes are not set or empty.
-}
+    NIError err = kNIError_Success;
 
-bool VariantType::IsAttributeMapNullOrEmpty(VariantTypeRef variant)
-{
-    return variant->_attributeMap == nullptr || variant->_attributeMap->size() == 0;
+    auto variantValue = static_cast<VariantDataRef *>(pData);
+    if (*variantValue != nullptr) {
+        VIREO_ASSERT((*variantValue)->IsUninitializedVariant());
+    } else {
+        if (pattern == nullptr) {
+            pattern = this;
+        }
+        VariantDataRef newVariant = VariantData::New(pattern);
+        *static_cast<VariantDataRef*>(variantValue) = newVariant;
```

Comment:
I think we shouldn't need the static_cast here. *variantValue = newVariant, should work. Is that not the case?